android/app/src/main/java/app/organicmaps/car/CarAppSession.java

```diff
@@ -101,91 +104,86 @@ public Screen onCreateScreen(@NonNull Intent intent)
       return new RequestPermissionsScreen(getCarContext(), this::onLocationPermissionsGranted);
     }
 
+    if (mDisplayManager.isDeviceDisplayUsed())
+    {
+      mScreenManager.push(mapScreen);
+      mSurfaceRenderer.disable();
+      onStop(this);
+      return new MapPlaceholderScreen(getCarContext());
+    }
+
     return mapScreen;
   }
 
   @Override
   public void onNewIntent(@NonNull Intent intent)
   {
     Logger.d(TAG, intent.toString());
-    // TODO (AndrewShkrob): This logic will need to be revised when we introduce support for adding stops during navigation or route planning.
-    // Skip navigation intents during navigation
-    if (RoutingController.get().isNavigating())
-      return;
-
     IntentUtils.processIntent(getCarContext(), mSurfaceRenderer, intent);
   }
 
   @Override
   public void onCreate(@NonNull LifecycleOwner owner)
   {
     Logger.d(TAG);
-    final DisplayManager displayManager = DisplayManager.from(getCarContext());
-    displayManager.addListener(DisplayType.Car, this);
-    getLifecycle().addObserver(displayManager.getObserverFor(DisplayType.Car));
+    mDisplayManager = DisplayManager.from(getCarContext());
+    mDisplayManager.addListener(DisplayType.Car, this);
     init();
   }
 
   @Override
   public void onStart(@NonNull LifecycleOwner owner)
   {
     Logger.d(TAG);
-    restoreRoute();
-    if (DisplayManager.from(getCarContext()).isCarDisplayUsed())
+    if (mDisplayManager.isCarDisplayUsed())
```

Comment:
I don't want to make these changes in this PR. I don't really like this idea now and It will require a lot of testing. Let's make it later. You can create an issue for this.